### ISSUE TITLE
`saw-rustc` doesn't work with struct newtypes

### ISSUE THREAD

**Issue by RyanGlScott** [GaloisInc/mir-json#53]
Surprisingly, `saw-rustc` will fail to compile this program:

```rs
pub struct S(u32);

pub fn f(s: S) -> u32 {
    match s {
        S(x) => x,
    }
}
```
```
$ saw-rustc test.rs
test build - extract output path - ["rustc", "test.rs", "--test", "--target", "x86_64-unknown-linux-gnu", "--cfg", "crux", "-L", "/home/ryanscott/Documents/Hacking/Rust/../Haskell/crucible/crux-mir/rlibs", "--sysroot", "/home/ryanscott/Documents/Hacking/Rust/../Haskell/crucible/crux-mir/rlibs"]
test build - ["rustc", "test.rs", "--target", "x86_64-unknown-linux-gnu", "--cfg", "crux", "-L", "/home/ryanscott/Documents/Hacking/Rust/../Haskell/crucible/crux-mir/rlibs", "--sysroot", "/home/ryanscott/Documents/Hacking/Rust/../Haskell/crucible/crux-mir/rlibs", "--cfg", "crux_top_level", "--crate-type", "rlib"]
error: #[test] can only be applied to functions
 --> test.rs:1:1
  |
1 | pub struct S(u32, u32);
  | ^^^^^^^^^^^^

note: Emitting MIR for test/0601a2ff::f

note: Emitting ADT definition for test/0601a2ff::S::_adtb7803c2264daf0ec[0]

error: aborting due to previous error

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ErrorGuaranteed(())', src/bin/mir-json-rustc-wrapper.rs:252:65
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This is surprising to me, since `saw-rustc` _does_ compile this variant of the program:

```rs
pub struct S {
    pub x: u32,
}

pub fn f(s: S) -> u32 {
    match s {
        S { x } => x,
    }
}
```

I wonder if there is a `DefKind` case that we are missing here:

https://github.com/GaloisInc/mir-json/blob/3b538589852282a752e517b69fc61b627cd6d4bd/src/analyz/mod.rs#L793

**Comment by RyanGlScott** [comment]
Some debug printing:

```diff
diff --git a/src/analyz/mod.rs b/src/analyz/mod.rs
index dbaef1d..8028a7b 100644
--- a/src/analyz/mod.rs
+++ b/src/analyz/mod.rs
@@ -790,6 +790,7 @@ fn init_instances_from_tests(ms: &mut MirState, out: &mut impl JsonOutput) -> io
             continue;
         }
 
+        println!("{:?}", tcx.def_kind(def_id));
         if !([DefKind::Fn, DefKind::AnonConst].contains(&tcx.def_kind(def_id))) {
             tcx.sess.span_err(
                 tcx.def_span(def_id),
```

Reveals that `S` in the failing program above gives rise to a `Ctor(Struct, Fn)` `DefKind`. For some reason, this doesn't happen when `S` is not defined as a newtype.

**Comment by RyanGlScott** [comment]
Fixed in #54.
